app/views/online-class/components/AvailableTime.vue

```diff
@@ -84,21 +84,22 @@ export default {
     getDates () {
       const dates = new Set()
       this.allTimes.forEach(t => {
-        dates.add(t.clone().tz(this.userTz).format('YYYY-MM-DD'))
+        dates.add(t.format('YYYY-MM-DD'))
       })
       return Array.from(dates)
     },
-    serverDate (date, time) {
-      return moment.tz(date, this.userTz).set({
+    toServerDateTime (date, time) {
+      const serverTime = moment.tz(date, this.userTz).set({
         hour: Math.floor(time),
         minute: Math.floor((time - Math.floor(time)) * 60),
         second: 0,
         millisecond: 0
-      }).tz(this.serverTz).format('YYYY-MM-DD')
+      }).tz(this.serverTz)
+      return [serverTime.format('YYYY-MM-DD'), serverTime.get('hour') + serverTime.get('minute') / 60]
```

Comment:
are we sending time like 10.16667 if hours is 10 and minutes is 10. Any reason for sending data like that instead of sending serverTime itself?

Reply:
it's my fault. i think Date object is not so stable in js since it doesn't include timezone by default so I use hours int at first. then I find the timezone offset can be half an hour or 15 mins. so I update the int hour to float hour. thankfully normal timezone only have 0.5 or 0.25. maybe some places uses 10mins offset? not a big problem I think